mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+// mpags-cipher.cpp
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main( int argc, char* argv[]){
+
+    const std::vector<std::string> cmdLineArgs{argv, argv+argc};
+
+    // Creating a boolean variable to tell us when help us requested.
+    bool helpRequested{false};
+
+    // Creating a boolean variable to tell me when the version is requested.
+    bool versionRequested{false};
+
+    // Creating a boolean to check if the input file was given and initialising the input file variable
+    bool input_fileGiven{false};
+    std::string input_file{""};
+
+    // Creating a boolean to check if the output file was given and initialising the output file variable
+    bool output_fileGiven{false};
+    std::string output_file{""};
+
+    // Looking through the command line arguments, excluding the first one.
+    for (std::size_t i{1}; i < cmdLineArgs.size(); i++){
+
+        //std::cout <<  " The argument being considered now is " << cmdLineArgs[i] << std::endl;
+        //std::cout <<  "The argument index is " << i << std::endl;
+
+        //Looking for whether help is requested
+        if((cmdLineArgs[i] == "--help") || (cmdLineArgs[i] == "-h") ){
+
+            helpRequested = true;
+
+        }
+
+        // Looking for whether the version is requested
+        if(cmdLineArgs[i] == "--version" ){
```

Comment:
This and subsequent if's should probably `if else`'s, or you should have `continue;` at the end of each, to avoid all possibilities being tested in each iteration of the loop.